### ISSUE TITLE
MS-TCN: Multi-Stage Temporal Convolutional Network for Action Segmentation

### ISSUE THREAD

**Issue by shuto-keio** [shuto-keio/paper_summary#19]
# めも
- oversegmentation：必要以上に細かいsegmentに分割してしまうこと
- 本論文では、oversegmentationを減らすことが第一目標
- 特徴
  - poolingを使用しない
  - multi stageなdilated convolution
  - over segmentationを防ぐためのsmooting loss
# 疑問点
2-stage目の入力に確率値のみを入力するのはどういう発想から？

# Abstruct
- 従来から使われているアプローチでは、フレームごとに*確率値*を計算し、高次元の時系列モデルに音死んでいる
- 近年で用いられるアプローチでは、時系列畳み込みを行う

- 時系列行動認識のためのmulti-stageなネットワークを提案
- それぞれのstageはdilated TCNを用いている
- stageをたどるごとに出力の確率値が洗練されていく

- 損失関数として、classification loss+smoothing loss
- smooting lossはover-segmentation errorを罰するような損失
- 50Salads, GTEA, Breakfast datasetでSOTA

# Introduction
- 従来のaction segmentationは2つのアプローチ
  - sliding window, Markov model  
  →制度が良いが、とても遅い
  -  temporal convolution  
  →時間方向の解像度が低い（？）

- Multi-Stage Temporal Convolutional Network(MS-TCN)の提案  
- stageを経るごとに、予測結果が精錬されていく

- dilated convolutionの特徴  
  少ないパラメータで広い受容野が得られる

# Related Work
- sliding window方式のtemporal segmentation[22, 11]

# proposed method
### Single-Stage TCN
- pooling, FCの不使用
- acausalでのTCN, kernel size=3
- 1x1 Conv→(dilated Conv → 1x1 Conv) x n(layer)

dilated Conv → 1x1 Conv の部分(skip connection有り)
![image](https://user-images.githubusercontent.com/38306927/61177851-2284cd80-a61a-11e9-9aa1-faa9a3c8ba9e.png)

- Receptive Field = 2^(l+1)-1

### Multi-Stage TCN
- motivated by [29, 18](human pose estimation)
![image](https://user-images.githubusercontent.com/38306927/61177852-26b0eb00-a61a-11e9-9143-5aeb14008b33.png)

### Loss Function
- 基本的には以下のような形で実験を行なっている
  - 交差エントロピー＋T-MSE
  - 交差エントロピー＋KL-Divergence
- 交差エントロピーはクラス確率も対してのロス
- T-MSE, KL-Divergenceはover-segmentationを減らすためのロス

##### 交差エントロピー
![image](https://user-images.githubusercontent.com/38306927/61177853-2e708f80-a61a-11e9-92ce-b9ea1303c968.png)

##### trancated mean squared loss
- 作者が提案した損失関数
- 前フレームと次フレームの間でクラス確率が大きく変化しないようにする
![image](https://user-images.githubusercontent.com/38306927/61177854-34ff0700-a61a-11e9-95de-09ca84efe177.png)

##### KL-Divergence
- 二つの確率分布間の距離を下げる働き
![image](https://user-images.githubusercontent.com/38306927/61177855-392b2480-a61a-11e9-8cba-17601a8fb6ac.png)

##### ロスはステージごとに加算される
![image](https://user-images.githubusercontent.com/38306927/61177857-3d574200-a61a-11e9-9b1f-70393b8bfd73.png)

# 実験概要
## Dataset
- 50Salads
  - 50videos
  - about 6.4 min per video
- GTEA
  - 28 videos
  - 調理者の頭につけられた画像で学習
- Breakfast
  - 1712 videos

## 評価指標
- FrameごとのAccuracy
- IoUに対してF1@{10, 25, 50}
- edit distance(レーベンシュタイン距離)
  - 一文字の挿入、削除、置換によって一方の文字列をもう一方の文字列に変形するのに必要な手順の最小回数(wikipediaより)
  - これを時系列方向のクラスラベルに対して行う
# 実験結果
## Stage数による影響
- stage1~stage4まではstageを増やすほど、F1, Edit, ACCが減少
- multi-stageにすることで、over segmentationの現象が図からもわかる

![image](https://user-images.githubusercontent.com/38306927/61177858-45af7d00-a61a-11e9-91d0-d35721d236a9.png)

## MS-TCNとDeeper S-TCNの比較
- 同じ層数の時、MS-TCNとS-TCNのどちらの精度が良いか
- 全ての項目においてMS-TCNの方が性能が良い
![image](https://user-images.githubusercontent.com/38306927/61177860-4b0cc780-a61a-11e9-8bf2-ad335ee5d7f3.png)

## KL-divergenceとT-MSEの比較
- T-MSEの方が性能が良い
![image](https://user-images.githubusercontent.com/38306927/61177862-506a1200-a61a-11e9-9604-88a1dd7b331f.png)

- T-MSEの方が小さい差でも大きく損失を取るため
![image](https://user-images.githubusercontent.com/38306927/61177863-552ec600-a61a-11e9-81b7-65bc814986a9.png)

## Passing Features to Higher Stages
- stageを重ねる際に次のstageに渡されるベクトルが、①確率値だけ、②最後の1D1Convの後の出力での比較

- 確率値を渡す方が性能が良い
  - 次のステージに特徴を渡すと、モデルがoversegmentaionを起こしてしまう
![image](https://user-images.githubusercontent.com/38306927/61177866-5a8c1080-a61a-11e9-977f-718de1bf851f.png)

## 時間方向の動画の解像度の比較
- 1fps vs 15fps
- この論文のベースは15fps
- 解像度が低いとoversegmentationが起きにくくなる、
![image](https://user-images.githubusercontent.com/38306927/61177869-5f50c480-a61a-11e9-8cde-4094a7192863.png)

## レイヤー数の比較
- layer数を0→10で増やしていくと性能が上がる
- 12の時にACCだけ下がるが、F値は向上
![image](https://user-images.githubusercontent.com/38306927/61177870-64157880-a61a-11e9-8658-541c52fe3556.png)

- 長時間の映像で性能が落ちる
  - 受容野が関係しているかも？
![image](https://user-images.githubusercontent.com/38306927/61177871-6972c300-a61a-11e9-99e1-320e57647a9c.png)

## fine-tuningの結果
![image](https://user-images.githubusercontent.com/38306927/61177874-7099d100-a61a-11e9-8d94-8576ccb2caa7.png)

## SOTAとの比較
- I3D: appearance + motion
- IDT(improved dense trajectories): only appearnce
- IDTの方が性能が良いため、appearance情報はあまり必要でない
![image](https://user-images.githubusercontent.com/38306927/61177875-755e8500-a61a-11e9-88f6-97550d9a3a0f.png)

# Conclusion
- oversegmentation errorを減少させることができた
  - smoothing loss
  - multi stage architecture
